staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go

```diff
@@ -104,6 +104,10 @@ func (f *FieldManager) Update(liveObj, newObj runtime.Object, manager string) (r
 			return nil, fmt.Errorf("failed to decode managed fields: %v", err)
 		}
 	}
+	// if managed field is still empty, skip updating managed fields altogether
```

Comment:
I think we can move that before `DecodeObject...`, since we don't need to decode in order to count

Reply:
Though if it's empty it's pretty fast to decode :-)